server/src/main/java/org/elasticsearch/common/settings/AddStringKeyStoreCommand.java

```diff
@@ -83,8 +85,19 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
 
         final char[] value;
         if (options.has(stdinOption)) {
-            BufferedReader stdinReader = new BufferedReader(new InputStreamReader(getStdin(), StandardCharsets.UTF_8));
-            value = stdinReader.readLine().toCharArray();
+            try (BufferedReader stdinReader = new BufferedReader(new InputStreamReader(getStdin(), StandardCharsets.UTF_8));
+                 CharArrayWriter writer = new CharArrayWriter()) {
+                int charInt;
+                while ((charInt = stdinReader.read()) != -1) {
+                    if ((char) charInt == '\r' || (char) charInt == '\n') {
+                        break;
+                    }
+                    writer.write((char) charInt);
+                }
+                value = writer.toCharArray();
+            } catch (IOException e) {
+                throw new UserException(ExitCodes.DATA_ERROR, e.getMessage());
```

Comment:
UserException is meant to be something clear for users which does not warrant a stack trace. Is the raw IOException message actually clear? What possible error scenarios exist?

Reply:
You're right, this was an oversight. IOException can be thrown by `read()` in the ByteBuffer, so I added a generic message and added the IOException as cause in the UserException